pkg/vmprovider/providers/vsphere/contentlibrary/content_library_cache.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package contentlibrary
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi/ovf"
+)
+
+type Cache struct {
+	mu      sync.Mutex
+	items   map[string]*itemCache
+	maxSize int
+}
+
+type itemCache struct {
+	ovfEnvelope    ovf.Envelope
+	contentVersion string
+	lastAccessTime int64
+}
+
+// NewCache creates a new content library cache with the given item expiration.
+// It also starts a goroutine to clean up expired cache items at the given check interval.
+func NewCache(itemExpiration, expirationCheckInterval time.Duration, maxSize int) *Cache {
+	c := &Cache{items: make(map[string]*itemCache), maxSize: maxSize}
+
+	// The following go routine cleans up expired cache items at the given check interval.
+	go func() {
+		itemExpirationSec := int64(itemExpiration.Seconds())
+		for range time.Tick(expirationCheckInterval) {
+			c.mu.Lock()
+			for key, item := range c.items {
+				if time.Now().Unix()-item.lastAccessTime > itemExpirationSec {
```

Comment:
This is just for removing items that have been cached for a while (longer than the itemExpirationSec) so that we can free some space in the cache. Otherwise, there is no way to add more items after it reaches max size.